packages/firebase_dart/lib/src/database/impl/treestructureddata/treestructureddata.dart

```diff
@@ -82,7 +82,9 @@ class Snapshot extends UnmodifiableMapBase<Name, Snapshot> {
 
     var v = isLeaf
         ? value!.toJson()
-        : map((k, v) => MapEntry(k.asString(), v.toJson(exportFormat)));
+        : keys.isList
```

Comment:
This change must iterate over the keys a few times for maps, however, the isList check should bail out early if any key is not an integer, which is likely.